### PR TITLE
Add option to disable automatic reporting of queue depth metrics

### PR DESCRIPTION
**Pull request by MatiasFernandez**
# Problem

The current approach for most sidekiq metrics is to report them when a job is processed. While this is good for some metrics is less than ideal for others. If you want to monitor the status of your queues you will need queue size and queue latency metrics even when no jobs are being processed. With the current approach that is not possible so we cannot set alerts when the queues are not being consumed.
# Solution

We have to disable the automatic reporting of queue size and latency metrics during the processing of a job and also add a new way of reporting this metric that doesn't depend on sidekiq actually working.

For that purpose I added a configuration attribute to manually disable queue depth metrics (size and latency). I also added a method that will report the status of all queues when it's called. That method should be called by a [clock process](https://devcenter.heroku.com/articles/scheduled-jobs-custom-clock-processes) in the main app

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,15 +50,43 @@ require 'nightwing/extensions/redis' # redis gem required
...
+| debug            | Enable for verbose logging     | no        | false             |
+| logger           | Logger instance for debug mode | no        | Nightwing::Logger |
+| disabled_metrics | Metrics that are disabled      | no        | Empty array       |
 
 When debug mode is turned on, Nightwing will output the metrics into a parsable format. The output destination is determined by the logger. If no logger is given then we send the debugging output to STDOUT.
 
+## Disabling automatic metrics
+
+The current approach of this gem is to report metrics in a custom sidekiq middleware. For some metrics this can be 
+less than ideal because you won't report the metric if no jobs are being processed. This is specially true for
```
Comment (pipt) [inline README.md]: specially should be especially.

File: spec/nightwing/sidekiq/queue_stats_spec.rb [hunk 2]
```diff
@@ -1,40 +1,69 @@
...
+        subject.call(nil, nil, fake_queue.name) do
           # beep
         end
       end
+
+      context "and queue depth metrics are disabled" do
+        let(:disabled_metrics) { [:queue_depth] }
+
+        it "does not report them" do
+          expect(subject.client).not_to receive(:measure).with("sidekiq.fake_queue.size", any_args).and_call_original
```
Comment (pipt) [inline spec/nightwing/sidekiq/queue_stats_spec.rb]: As we're asserting `not_to receive`, is the `and_call_original` needed here (and in the context below)?

**Comment by MatiasFernandez** [comment]
Both comments addressed @pipt

**Comment by MatiasFernandez** [comment]
Pinging @timminkov to be in the loop of this change

**Comment by timminkov** [comment]
👍

**Pull request merged**